tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+
+To help you get the most from your Ubuntu experience, this tutorial will walk
+you through managing the software on your Ubuntu machine. Although we'll be
+using Ubuntu Server as our example, this tutorial is equally applicable to the
+Ubuntu Desktop.
+
+There are two types of software found in Ubuntu: **Debian packages** and
+**snaps** -- we will learn about both!
+
+## Create the virtual machine
+
+To avoid making changes to your computer we will set up a virtual machine (VM)
+to launch our server and run the commands.
+
+Multipass is great for quickly creating Ubuntu virtual machines, so we will use
+that. On Ubuntu, you can install Multipass by running the following command in
+your terminal:
+
+```bash
+sudo snap install multipass
+```
+
+Multipass can also be installed on Windows, Mac and other Linux distributions
+[using these instructions](https://multipass.run/docs/tutorial#install-multipass).
+
+Once you have installed Multipass, it is straightforward to launch a new VM.
+Let us launch our VM using the 24.04 LTS release (`noble`), and let's also give
+it the name `tutorial` using the following command. The default VM created with
+this command will need **5 GiB of disk space**, and **1 GiB of memory**:
+
+```bash
+multipass launch noble --name tutorial
+```
+
+Multipass will download the most recent daily **image** and create the VM for
+us. It may take a little time, depending on the speed of your internet
+connection.
+
+The **image** is the collection of files we need to install and run Ubuntu. We
+don't need to specify "server" or "desktop" anywhere in our command, because
+the image is the same for both. The only difference between Ubuntu Server and
+Ubuntu Desktop is the subset of software packages we use from the image - we
+will see this later!
+
+Now we can access the VM by running:
+
+```bash
+multipass shell tutorial
+```
+
+We will get a "Welcome to Ubuntu 24.04 LTS" message. Notice that when we run
+this command, the terminal username changes to `ubuntu` and the hostname
+changes to `tutorial`:
+
+```bash
+ubuntu@tutorial
+```
+
+This shows that we are inside the VM, where we will run all of our commands. 
+
+## Updating the system with APT
+
+The first thing we always want to do with a new system (whether a VM or a
+"live" machine) is to run an update to make sure we have the latest versions
+of all the default software. 
+
+Debian packages, commonly referred to as **"debs"** are the standard software
+in Ubuntu. They can be identified by the `.deb` file extension and usually come
+in a pre-compiled binary format (so you don't need to compile it yourself).
+
+Your system software (and other deb packages) are handled through the
+[Advanced Packaging Tool](https://ubuntu.com/server/docs/package-management#advanced-packaging-tool)
+(APT). APT is a package *repository*; it contains both the **database** of the
+packages available in Ubuntu, and it contains the **packages** themselves.
```

Comment:
I'm not sure that `pip` use is extensive/widespread enough to make the comparison meaningful (especially because APT is the default in Ubuntu), but I can/should probably make the package management aspect more explicit. I'll have a fiddle and see what improvements I can make.